sdk/eventhub/event-hubs/src/eventHubReceiver.ts

```diff
@@ -200,214 +182,234 @@ export class EventHubReceiver extends LinkEntity {
+      this._onMessage(receivedEventData);
+    } else {
+      this._internalQueue.push(receivedEventData);
+    }
+  }
 
-    this._onSessionClose = async (context: EventContext) => {
-      const receiver = this._receiver || context.receiver!;
-      const sessionError = context.session && context.session.error;
-      if (sessionError) {
-        log.error(
-          "[%s] 'session_close' event occurred for receiver '%s' with address '%s'. " +
-            "The associated error is: %O",
-          this._context.connectionId,
-          this.name,
-          this.address,
-          sessionError
-        );
-      }
+  private _onAmqpError(context: EventContext): void {
+    const amqpReceiver = this._receiver || context.receiver;
+    if (!amqpReceiver) {
+      return;
+    }
 
-      if (receiver && !receiver.isSessionItselfClosed()) {
-        if (!this.isConnecting) {
-          log.error(
-            "[%s] 'session_close' event occurred on the session of receiver '%s' with " +
-              "address '%s' and the sdk did not initiate this. Hence calling detached from the " +
-              "_onSessionClose() handler.",
-            this._context.connectionId,
-            this.name,
-            this.address
-          );
-          await this.onDetached(sessionError);
-        } else {
-          log.error(
-            "[%s] 'session_close' event occurred on the session of receiver '%s' with " +
-              "address '%s' and the sdk did not initiate this. Moreover the receiver is already " +
-              "re-connecting. Hence not calling detached from the _onSessionClose() handler.",
-            this._context.connectionId,
-            this.name,
-            this.address
-          );
-        }
-      } else {
-        if (this._abortSignal) {
-          this._abortSignal.removeEventListener("abort", this._onAbort);
-        }
-        log.error(
-          "[%s] 'session_close' event occurred on the session of receiver '%s' with address " +
-            "'%s' because the sdk initiated it. Hence not calling detached from the _onSessionClose" +
-            "() handler.",
-          this._context.connectionId,
-          this.name,
-          this.address
-        );
-      }
-    };
+    const amqpError = amqpReceiver.error;
+    if (!amqpError) {
+      return;
+    }
+
+    const error = translate(amqpError);
+    log.error(
+      "[%s] An error occurred for Receiver '%s': %O.",
+      this._context.connectionId,
+      this.name,
+      error
+    );
+
+    if (!amqpReceiver.isItselfClosed() && this._onError) {
+      log.error(
+        "[%s] Since the user did not close the receiver " +
+          "we let the user know about it by calling the user's error handler.",
+        this._context.connectionId
+      );
+      this._onError(error);
+    } else {
+      log.error(
+        "[%s] The receiver was closed by the user." +
+          "Hence not notifying the user's error handler.",
+        this._context.connectionId
```

Comment:
This else block is also reached when there is `_onError`
In that case, the log message here is misleading.

Would suggest the below
```javascript
if (amqpReceiver.isItselfClosed()) 
 // log 
 return;
}
if (this._onError) {
 // log
  this._onError(error);
}
```

Same for the session error